src/db-backend/tests/test_harness/mod.rs

```diff
@@ -0,0 +1,664 @@
+//! Shared test harness for DAP-based integration tests
+//!
+//! This module provides abstractions for:
+//! - Building and recording test programs
+//! - Managing DAP client connections
+//! - Running flow tests across different language versions
+//!
+//! # Design
+//!
+//! The harness separates concerns into:
+//! - `TestRecording`: A compiled and recorded trace that can be reused across tests
+//! - `DapTestClient`: A wrapper around the DAP client with test-friendly helpers
+//! - `FlowTestCase`: Configuration for running flow-based tests
+
+use db_backend::dap::{self, DapClient, DapMessage, LaunchRequestArguments};
+use db_backend::task::{CtLoadFlowArguments, FlowMode, Location};
+use db_backend::transport::DapTransport;
+use serde_json::json;
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+use std::io::{self, BufRead, BufReader, ErrorKind, Write};
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::path::{Path, PathBuf};
+use std::process::{Child, Command, Stdio};
+use std::time::{Duration, Instant};
+use std::thread;
+
+/// Supported languages for test programs
```

Comment:
we do have an existing `Lang` enum, and extension helpers for it somewhere, really better to reuse it in the future: probably not a problem for the test, just a general note

Reply:
`Lang` is in lang.rs, we have also `lang_from_context` (path + trace kind) , but the extension thing is hardcoded for now in flow_preloader(and we have helpers in nim, not rust for it, sorry)